src/Sentry.DiagnosticSource/Internals/DiagnosticSource/SentrySqlListener.cs

```diff
@@ -76,6 +84,15 @@ private void AddSpan(SentrySqlSpanType type, string operation, string? descripti
                     TryGetQuerySpan(scope, queryId) is { } querySpan)
                 {
                     span = querySpan;
+
+                    if (span.ParentSpanId == scope.Transaction?.SpanId &&
+                        span.Extra[ConnectionExtraKey] is Guid spanConnectionId &&
```

Comment:
Lets assume the value could have been removed from the dictionary and be safe when reading:
```suggestion
                        span.Extra.TryGetValue("ConnectionExtraKey", out var spanConnectionId) && spanConnectionId is Guid && spanConnectionId &&
```